File: seungyeop/22_12_16_02.java
//1. 2진수로 만든다. -> 2로 나누면서 2번까지 진행할 경우 2진수의 끝자리부터 넣어줘야 한다.
    //-> 왼쪽으로 넣어줘야 한다. -> + 연산자를 쓰면 가볍게 붙일 수 있음
//2. 둘 중 하나만 1이어도 #을 넣어준다.

class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        for (int i = 0; i < n; i++) { //라인 별 특징
            String temp = ""; //라인마다 새로 생성해주는 임시 저장소
            for (int j = 1; j <= n; j++) {
                if (arr1[i] % 2 == 1 || arr2[i] % 2 == 1) { //두 수의 나머지 중 1개 이상 1인 경우 #추가
                    temp = "#" + temp;
                }
                else {
                    temp = " " + temp;
                }
                arr1[i] /= 2;
                arr2[i] /= 2;
            }
            answer[i] = temp;   //정답에 넣어주기
        }
        return answer;
    }
}

File: seungyoon/22_12_16_02.py
def solution(n, arr1, arr2):
    비밀지도1 = []
    비밀지도2 = []
    
    # 비밀지도들을 2진수 값으로 변환
    # zfill 함수는 문자열을 특정 길이로 맞추기 위해 앞에 0을 추가해주는 함수
    # format(정수,"b")는 정수를 2진수로 바꿔주는 함수
    for i in range(n):
        지도1 = format(arr1[i],"b").zfill(n)
        지도2 = format(arr2[i],"b").zfill(n)
        진짜지도 = format(arr1[i],"b").zfill(n)
        비밀지도1.append(지도1)
        비밀지도2.append(지도2)
    
    # 2진수로 변환된 비밀지도 두개를 대조하여 진짜지도 만들기
    A = ""
    for i in range(n):
        A += "*"   
        for j in range(n):
            if 비밀지도1[i][j] == "1" or 비밀지도2[i][j] == "1":
                A += "#"
            elif 비밀지도1[i][j] == "0" and 비밀지도2[i][j] == "0":
                A += " "
    진짜지도 = A.split("*")
        
    return 진짜지도[1:] # 0인덱스에는 공백이 있으므로 1인덱스부터 시작
